bluetooth.go

```diff
@@ -262,8 +262,8 @@ func (d *Device) hci_read(b []byte) (uint32, error) {
 	if err != nil {
 		return 0, err
 	}
-	length := uint32(b[0]) | uint32(b[1])<<8 | uint32(b[2])<<16
-	hciLength := length + 4 // Add 3 bytes for SDIO header, plus 1 for packet type
+	length := 1 + (uint32(b[0]) | uint32(b[1])<<8 | uint32(b[2])<<16) // plus 1 for packet type
+	hciLength := length + 3                                           // plus 3 bytes for SDIO header
```

Comment:
After merging the previous PR it did come across my mind there maye have been a bug because we were reading round(length+4) bytes from the ring buffer and only signalling we were reading length+4.

This PR seems to suppose a different thing though:
- we still read round(length+4) bytes from the buffer
- We return reading lenghth+1 bytes only, thus trimming the last 3 to 6 bytes (depending on  rounding)

There seems to be a misconception in our understanding of the hci. How can trimming the last read bytes work?

Reply:
This PR seems to presume the SDIO header is at the end since it adds it to read it and discard it.

Reply:
"nonetheless it moves" however, if you will. It works consistently with the various samples.

Reply:
> We return reading lenghth+1 bytes only, thus trimming the last 3 to 6 bytes (depending on rounding)

I think that is correct, which is why the adjustment to the returned length is needed as the trailing bytes are all zeros and not valid data anyhow.